src/components/pages/EditTransactionPage.js

```diff
@@ -0,0 +1,55 @@
+import React, { useState, useEffect } from 'react'
+import TransactionFormInput from '../TransactionFormInput'
+import ToggleSwitch from '../ToggleSwitch'
+import ReturnToOverviewButton from '../ReturnToOverviewButton'
+import styled from 'styled-components/macro'
+import { editTransactionEntry, getTransactionEntry } from '../utils/services'
+import { useParams } from 'react-router-dom'
+
+export default function EditTransactionPage() {
+  const { id } = useParams()
+  const [selected, setSelected] = useState(false)
+  const [transaction, setTransaction] = useState({})
+
+  useEffect(() => {
+    getTransactionEntry(id).then(setTransaction)
+  }, [id])
+
+  useEffect(() => {
+    setSelected(transaction.type === 'income')
+  }, [transaction])
+
+  return (
+    <PageStyled>
+      <ToggleSwitch selected={selected} onToggle={handleToggle} />
+      <TransactionFormInput
+        onSave={onEditUpdateTransactionEntry}
+        notes={transaction.notes}
+        value={transaction.value?.toFixed(2).replace('.', ',')}
+        tag={transaction.tag}
+      />
+      <ReturnToOverviewButton />
+    </PageStyled>
+  )
+  function handleToggle() {
+    setSelected(!selected)
+  }
+
+  function onEditUpdateTransactionEntry(value, notes, tag) {
+    const updatedTransaction = {
+      type: selected ? 'income' : 'spending',
+      value: parseFloat(String(value).replace(',', '.')),
+      notes: notes,
+      tag: tag,
```

Comment:
I think when you don't change the object key you can also write them like this 
```suggestion
      notes,
      tag,
```

Reply:
Good point. In this case, I decided to leave them in to clarify that there is data that the user can update, and data that will remain unchanged.